### PR TITLE
Update docs with database user auto provisioning modes

### PR DESCRIPTION
**Pull request by gabrielcorado**
#32792 introduced a new role option to configure if provisioned database users should be kept (disabled) or dropped. This PR updates the auto-provisioning docs to add this new option.

Note: `create_db_user` will still work (`true = keep` and `false = off`), but it is preferred that users specify the mode (similar to the `create_host_user_mode` and `create_host_user` options).

### CODE REVIEW

File: docs/pages/includes/role-spec.mdx [hunk 1]
```diff
@@ -102,7 +102,7 @@ spec:
     #          and off (disable host user creation)
     create_host_user_mode: drop
     # Controls whether this role requires automatic database user provisioning.
-    create_db_user: true
+    create_db_user_mode: keep
```
Comment (greedy52) [inline docs/pages/includes/role-spec.mdx:105]: I think we should add comments explaining the options.

File: docs/pages/access-controls/reference.mdx [hunk 2]
```diff
@@ -120,6 +120,7 @@ The table below documents the behavior of each option if multiple roles are assi
 | `pin_source_ip` | Enable source IP pinning for SSH certificates. **Note:** IP pinning is currently in Preview mode | Logical "OR" i.e. evaluates to "yes" if at least one role requires session termination |
 | `cert_extensions` | Specifies extensions to be included in SSH certificates | |
 | `create_host_user_mode` | Allow users to be automatically created on a host |  Logical "AND" i.e. if all roles matching a server specify host user creation (`off`, `drop`, `keep`), it will evaluate to the option specified by all of the roles. If some roles specify both `drop` or `keep` it will evaluate to `keep`|
+| `create_db_user_mode` | Allow [database user auto provisioning](../database-access/auto-user-provisioning.mdx) | Logical "OR" i.e. if any role allows database user auto-provisioning, it's allowed |
```
Comment (greedy52) [inline docs/pages/access-controls/reference.mdx:123]: if possible, list all the options here.

**Review by greedy52 — APPROVED** [review]

**Review by greedy52 — COMMENTED** [review]
BTW, could you also align these text:

https://github.com/gravitational/teleport/blob/a9c31826ed42d049c5f7faf78807b84cd8c06eee/docs/pages/database-access/auto-user-provisioning/postgres.mdx#L72-L74

https://github.com/gravitational/teleport/blob/a9c31826ed42d049c5f7faf78807b84cd8c06eee/docs/pages/database-access/auto-user-provisioning/mysql.mdx#L73-L75

Thanks!!

**Comment by gabrielcorado** [comment]
@greedy52 I've removed those paragraphs from the pages since it is already mentioned in the "Create Teleport role" section.

File: docs/pages/includes/database-access/auto-user-provisioning-common-teleport-role.mdx [hunk 3]
```diff
@@ -30,7 +30,8 @@ The available provisioning modes are:
 
 * `off`: Disables user provisioning.
 
-* `keep`: Enables user provisioning and disables users at session end.
+* `keep`: Enables user provisioning and disables users at session end. session.
```
Comment (ptgott) [inline docs/pages/includes/database-access/auto-user-provisioning-common-teleport-role.mdx:33]:
```suggestion
* `keep`: Enables user provisioning and disables users at session end.
```

File: docs/pages/access-controls/reference.mdx [hunk 4]
```diff
@@ -120,7 +120,7 @@ The table below documents the behavior of each option if multiple roles are assi
 | `pin_source_ip` | Enable source IP pinning for SSH certificates. **Note:** IP pinning is currently in Preview mode | Logical "OR" i.e. evaluates to "yes" if at least one role requires session termination |
 | `cert_extensions` | Specifies extensions to be included in SSH certificates | |
 | `create_host_user_mode` | Allow users to be automatically created on a host |  Logical "AND" i.e. if all roles matching a server specify host user creation (`off`, `drop`, `keep`), it will evaluate to the option specified by all of the roles. If some roles specify both `drop` or `keep` it will evaluate to `keep`|
-| `create_db_user_mode` | Allow [database user auto provisioning](../database-access/auto-user-provisioning.mdx) | Logical "OR" i.e. if any role allows database user auto-provisioning, it's allowed |
+| `create_db_user_mode` | Allow [database user auto provisioning](../database-access/auto-user-provisioning.mdx) (`off`, `keep`, `best_effort_drop`) | Logical "OR" i.e. if any role allows database user auto-provisioning, it's allowed |
```
Comment (ptgott) [inline docs/pages/access-controls/reference.mdx:123]: Is there a way to describe each possible value here?

Comment (gabrielcorado) [reply docs/pages/access-controls/reference.mdx]: I've added the same description as the role spec.

**Review by ptgott — APPROVED** [review]

**Pull request merged**